### PR TITLE
fix(core): avoid reading empty .config.json, upgrade cosmiconfig@v9.0.0

### LINKED ISSUE #4061: fix(core): empty .config.json file in project root gets picked up as lerna config
This is the same issue that lerna-lite had here: https://github.com/lerna-lite/lerna-lite/issues/729

### Describe the bug

If I have an empty `.config.json` file in my project root then lerna will crash because it assumes that the config file was for itself and not for something else (and it tries to parse it and then crash because the config property is undefined)

IMO, since lerna is a tool that is commonly used as part of tooling of projects that have their own config files, the tool shouldn't assume that a generic name such as .config.json is a config file that is intended for itself (now that I think about it my app also shouldn't assume that TBH, but maybe it has a slightly stronger claim to "squat" on a generic name like this).

### Expectation

TLDR: The default search path for config files should consist of files that have lerna in their names, not the generic ones. This is just my opinion though, not a strongly held one either.

Or if you think that it's okay to assume `.config.json` belonging to lerna, that's fine by me, but then it should at least validate the config and error out with something more obvious explaining that the config was missing certain (mandatory) properties from it.

### Reproduction

```sh

cd some-lerna-project-root-dir
touch .config.json
yarn lerna run 'echo hello'
```

Which will then crash with:

```sh
lerna notice cli v8.1.7
lerna ERR! TypeError: Cannot use 'in' operator to search for 'extends' in undefined
lerna ERR!     at applyExtends (/home/peter/a/cacti/node_modules/lerna/dist/index.js:2429:17)
lerna ERR!     at Object.transform (/home/peter/a/cacti/node_modules/lerna/dist/index.js:2694:28)
lerna ERR!     at load (/home/peter/a/cacti/node_modules/lerna/node_modules/cosmiconfig/dist/ExplorerSync.js:20:32)
lerna ERR!     at emplace (/home/peter/a/cacti/node_modules/lerna/node_modules/cosmiconfig/dist/util.js:12:20)
lerna ERR!     at ExplorerSync.load (/home/peter/a/cacti/node_modules/lerna/node_modules/cosmiconfig/dist/ExplorerSync.js:23:42)
lerna ERR!     at ExplorerSync.search (/home/peter/a/cacti/node_modules/lerna/node_modules/cosmiconfig/dist/ExplorerSync.js:30:33)
lerna ERR!     at #resolveLernaConfig (/home/peter/a/cacti/node_modules/lerna/dist/index.js:2698:27)
lerna ERR!     at new _Project (/home/peter/a/cacti/node_modules/lerna/dist/index.js:2560:78)
lerna ERR! ENOPROJECT Lerna Project not initialized!
/home/peter/a/cacti/node_modules/lerna/dist/index.js:3181
          throw new ValidationError("ENOPROJECT", "Lerna Project not initialized!");
                ^

ValidationError: Lerna Project not initialized!
    at get project [as project] (/home/peter/a/cacti/node_modules/lerna/dist/index.js:3181:17)
    at /home/peter/a/cacti/node_modules/lerna/dist/index.js:3158:35 {
  prefix: 'ENOPROJECT'
}

Node.js v20.11.0
```

### Used Package Manager

yarn@4.3.1

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
{
  "version": "2.0.0-rc.3",
  "npmClient": "yarn",
  "command": {
    "version": {
      "message": "chore(release): publish"
    }
  },
  "changelogPreset": "angular"
}
```

</p></details>

<details><summary>lerna-debug.log</summary><p>

`N/A`

</p></details>

### Environment

```sh
$ npx lerna info
lerna notice cli v8.1.7

 Environment info:

  System:
    OS: Linux 5.10 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (24) x64 12th Gen Intel(R) Core(TM) i7-12850HX
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    Yarn: 4.3.1 - ~/.nvm/versions/node/v20.11.0/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.0/bin/npm
  Utilities:
    Git: 2.34.1 - /usr/bin/git
```

### PR DESCRIPTION
**Pull request by petermetz**
The credit should go to @ghiscoding - I'm just back-porting his fix to the same bug in lerna-lite to lerna here.

## Description

The newer version of cosmicconfig does not suffer from the issue that
it defaults to looking for .config.json the way 8.x does.
This upgrade implicitly fixes the underlying issue that lerna has for the
same reason.

## Motivation and Context

Fixes #4061

lerna-lite used to have the same bug, you can read the full context here:
1. https://github.com/lerna-lite/lerna-lite/issues/729
2. https://github.com/cosmiconfig/cosmiconfig/issues/313
3. https://github.com/cosmiconfig/cosmiconfig/pull/311

## How Has This Been Tested?

`npm test && npm run integration`

The tests all passed, but for the integratin tests 2 of them failed with the logs pasted below. They were failing the same way on `main` prior to my changes so I'm assuming this to be a false positive but you tell me.

```sh
Summary of all failing tests
 FAIL  __tests__/lerna-publish-validation.spec.ts
  ● lerna publish exits with EBEHIND when behind upstream remote

    Command failed with exit code 128: git commit -m upstream change
    Author identity unknown

    *** Please tell me who you are.

    Run

      git config --global user.email "you@example.com"
      git config --global user.name "Your Name"

    to set your account's default identity.
    Omit --global to set the identity only in this repository.

    fatal: empty ident name (for <peter@C11-73JFIMZA8J6.dir.svc.accenture.com>) not allowed

      26 |   await fs.outputFile(path.join(cloneDir, "README.md"), "upstream change");
      27 |   await gitAdd(cloneDir, "-A");
    > 28 |   await gitCommit(cloneDir, "upstream change");
         |   ^
      29 |   await execa("git", ["push", "origin", "main"], { cwd: cloneDir });
      30 |
      31 |   // throws during interactive publish (local)

      at makeError (../node_modules/execa/lib/error.js:59:11)
      at handlePromise (../node_modules/execa/index.js:114:26)
      at Object.<anonymous> (__tests__/lerna-publish-validation.spec.ts:28:3)

 FAIL  __tests__/lerna-bootstrap-yarn.spec.ts
  ● lerna bootstrap --npm-client yarn

    Command failed with exit code 1: node /home/peter/a/lerna/packages/lerna/dist/cli.js bootstrap --npm-client yarn
    lerna notice cli v8.1.8
    lerna info ci enabled
    lerna info Bootstrapping 4 packages
    lerna info lifecycle package-4@1.0.0~preinstall: package-4@1.0.0
    lerna info Installing external dependencies
    lerna ERR! yarn install --mutex network:42424 --non-interactive exited 1 in '@integration/package-2'
    lerna ERR! yarn install --mutex network:42424 --non-interactive stdout:
    Unknown Syntax Error: Unsupported option name ("--mutex").

    $ yarn install [--json] [--immutable] [--immutable-cache] [--refresh-lockfile] [--check-cache] [--check-resolutions] [--inline-builds] [--mode #0]
    lerna ERR! yarn install --mutex network:42424 --non-interactive exited 1 in '@integration/package-2'
    lerna WARN complete Waiting for 2 child processes to exit. CTRL-C to exit immediately.

    > package-4@1.0.0 preinstall /tmp/dd16a3ecb8e9a0c03b35291c2f37dd47/package-4
    > exit 0

       9 |   const lerna = cliRunner(cwd);
      10 |
    > 11 |   const { stderr } = await lerna("bootstrap", "--npm-client", "yarn");
         |                      ^
      12 |   expect(stderr).toMatchInlineSnapshot(`
      13 | lerna notice cli __TEST_VERSION__
      14 | lerna info ci enabled

      at makeError (../node_modules/execa/lib/error.js:59:11)
      at handlePromise (../node_modules/execa/index.js:114:26)
      at Object.<anonymous> (__tests__/lerna-bootstrap-yarn.spec.ts:11:22)


Test Suites: 2 failed, 31 passed, 33 total
Tests:       2 failed, 83 passed, 85 total
Snapshots:   93 passed, 93 total
Time:        68.037 s, estimated 83 s
Ran all test suites.

——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Ran target integration for project integration and 6 task(s) they depend on (1m)

      With additional flags:
        --maxWorkers=2

   ✖  1/7 failed
   ✔  6/7 succeeded [4 read from cache]

View structured, searchable error logs at https://nx.app/runs/8xAbKEIHdj
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [] Chore (change that has absolutely no effect on users)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by JamesHenry** [comment]
@petermetz Thanks for this! I can't reproduce an issue, it seems to be specific to your environment.

What does `git config --global user.email` and `git config --global user.name` return for you on your machine?

**Comment by JamesHenry** [comment]
I made sure everything was all good on latest main here: https://github.com/lerna/lerna/pull/4071

So I am good to merge this in the meantime, thanks again!

**Comment by petermetz** [comment]
@JamesHenry Thank you very much for the review!

> What does git config --global user.email and git config --global user.name return for you on your machine?

Nothing - I set these parameters on a per-repo basis only. 

Now, with that said, if I run them without the `--global` flag then it would show my corporate email address for user.email (the one that is associated with my GitHub account - I double checked) and "Peter Somogyvari" for user.name. Not sure if this helps or not but I didn't want to just sidestep the question. :-)
